### ISSUE TITLE
Help wanted?

### ISSUE THREAD

**Issue by r2d2rigo** [LogicAndTrick/sledge#255]
I'm back to doing some minor modding on my spare time and came across this project - I think it's really great to have an open source version of Worldcraft/Hammer, and written in C# no less!

Are there any tasks up for grabs or areas where help is needed? Would love to contribute a bit with development.

**Comment by wetswan** [comment]
I'm interested as well. Like the project and if there are smaller issues I'd be happy to help.

**Comment by LogicAndTrick** [comment]
Thanks guys, I would love help in any form, whether it be bug fixes, features, performance improvements, translations into non-English, documentation, suggestions, etc. I've been working on a major rebuild in the default branch and its close to done but still missing a few things. Even just some feedback from people using the new build (from the simplification-project branch) would be very useful. Unfortunately I've broken an arm this past weekend so I wont be able to do much actual code work myself in the next month or two :(

I'm pretty poor at project management so I'm not very organised, but anyone who would like to get involved is more than welcome and I'll help out with whatever I can.

One little standalone thing that I would like to make is an editor (e.g. winforms) or an CSV import/export command line tool for the translation json files to make it easy for non-technical people to translate from English.

**Comment by r2d2rigo** [comment]
@LogicAndTrick aw sucks, hope you get well soon! I see that #238 has a list of features needed for the alpha/beta rebuilds - would you be able/willing to do a Trello board or set them up as GitHub issues with a help-wanted tag? 3-4 lines describing what is needed and how to do it would be enough to make them accessible to anyone that wants to collaborate.

Regarding localization, why reinvent the weel? There are existing tools like [Poedit](https://poedit.net/) that are easy to use - and you can pair it with regular [Gettext](https://github.com/neris/NGettext) to have a very straighforward implementation.

**Comment by TeamSpen210** [comment]
Well, the difference is that the JSON is hierarchical, so things are grouped into the different areas. What coukd be done is writing a simple utility to convert, and then people could translate in whatever version.

**Comment by DexterHaslem** [comment]
Hi there, I'd like to help out with the external translation editor tool, do you want it in a sub directory of sledge or its own repo?

**Comment by LogicAndTrick** [comment]
Thanks! I haven't forgotten about all you folks who would like to help but I'm still recovering so I can't do too much for a few more weeks at least. 

For the translation tool: Add it as a new project in the main repo, that's the easiest way to manage in my opinion. It could be either a winforms GUI or a command line csv converter depending on which you prefer to work on. 

Essentially it should use the English json files as the "master format" and allow translators to see the English text and convert it into the target language side-by-side and save it into the same format. 

One thing I'm not sure about is how the new translated files get back into the GitHub repo because it might not be reasonable to expect translators to use Git and create a PR. But we can work that out at a later point, the hardest part is creating the files in the first place. (I'm thinking maybe a small php web service that the app can post the files to, I don't have any ASP hosting so it would need to be php)

**Comment by Grocel** [comment]
You would have to be careful about the translation base, as English to <Language> is not safe from contextual problems. Better would be a context identifier with an English description to <Language>.

There are also platform made for crowd based translations such as crowdin.com.
Garry's Mod and Minecraft uses or did use such platforms and it seems it did a quite good job. Maybe that helps.

**Comment by goodDOS** [comment]
I can help translate into Chinese.

**Comment by JoelTroch** [comment]
@LogicAndTrick 
> One little standalone thing that I would like to make is an editor (e.g. winforms) or an CSV import/export command line tool for the translation json files to make it easy for non-technical people to translate from English.

Do you have a sample JSON file to show?

**Comment by LogicAndTrick** [comment]
Yep, each project has a Translations folder with them in, for example [this one](https://github.com/LogicAndTrick/sledge/blob/feature/simplification-project/Sledge.Shell/Translations/Sledge.Shell.en.json).

**Comment by DJPlaya** [comment]
Make a Post when there are News about the Translator or whatever is going to be used - Ill do the German Translations then

**Comment by LogicAndTrick** [comment]
Will do! The translator will be included in the next release, which I'm hoping will be done by this weekend or next weekend at the latest.

**Comment by LogicAndTrick** [comment]
@DJPlaya: The translator app is available now. A Discord member has partially done a German translation (about 70% complete) but he was moving on to other things and was requesting someone else to proof read and finish the translations. If you're still interested, it'd be great to get the German translation completed!

The partial translation is here: [Partial-Sledge-German-Translation.zip](https://github.com/LogicAndTrick/sledge/files/2533679/Partial-Sledge-German-Translation.zip)

To use it, unzip the contents to `%AppData%\Sledge` so you have a folder structure like `%AppData%\Sledge\Translations\[file names here].de.json`, and then launch Sledge and go to "Tools > Sledge Translator..." to start the translator app. Then you can just zip up the updated files from AppData again and drop them into a GitHub comment and I'll add them to the next release.

Thanks in advance! If anybody else is interested in translating, any language is welcome! I plan to eventually make a better way for translators to submit updates, but this is good enough for now I think.

**Comment by DJPlaya** [comment]
I wish i would have been knowing this earlyer ^^, but np, ill try to finish his work
